src/Datadog.Trace/FrameworkDescription.NetFramework.cs

```diff
@@ -8,15 +8,36 @@ namespace Datadog.Trace
 {
     internal partial class FrameworkDescription
     {
-        public static FrameworkDescription Create()
+        private static FrameworkDescription _instance = null;
+
+        public static FrameworkDescription Instance
+        {
+            get { return _instance ?? (_instance = Create()); }
+        }
+
+        private static FrameworkDescription Create()
         {
-            // .NET Framework
+            var osArchitecture = "unknown";
+            var processArchitecture = "unknown";
+            var frameworkDescription = "unknown";
+
+            try
+            {
+                osArchitecture = Environment.Is64BitOperatingSystem ? "x64" : "x86";
+                processArchitecture = Environment.Is64BitProcess ? "x64" : "x86";
+                frameworkDescription = GetNetFrameworkVersion();
```

Comment:
nit: Can this variable be renamed? Maybe `frameworkVersion` ?

Reply:
Done